### PR TITLE
[Metrics v2] Search apis

### LINKED ISSUE #41009: `/api/search` - return new metrics for `metric`

### LINKED ISSUE #41010: `/api/collection/:id/items` and `root` - return new metrics for `metric`

### PR DESCRIPTION
**Pull request by snoe**
Fixes #41009
Fixes #41010 

Update search and collection apis to return metrics, updates card related and revisions to support metrics.

### CODE REVIEW

File: src/metabase/related.clj [hunk 1]
```diff
@@ -159,20 +168,21 @@
   [card]
   (->> (t2/select Card
          :table_id (:table_id card)
+         :type [:in [:model :question]]
```
Comment (metamben) [inline src/metabase/related.clj:171]: I'm not sure this constraint is needed. Probably a product decision.

File: src/metabase/related.clj [hunk 2]
```diff
@@ -159,20 +168,21 @@
...
+  (->> (t2/select Card
          :table_id (:table_id card)
+         :type :metric
          :archived false)
        filter-visible
-       (m/find-first (comp #{(-> card :dataset_query :query :aggregation)}
-                           :aggregation
-                           :definition))))
+       (rank-by-similarity card)
+       (filter (comp pos? :similarity))))
```
Comment (metamben) [inline src/metabase/related.clj:185]: This is also a product question, IMO. I can imagine that now that metrics become first class citizens, we want to compare them against questions and models. E.g., people might write questions and models that are better modeled as metrics.

File: src/metabase/search/scoring.clj [hunk 3]
```diff
@@ -321,7 +322,11 @@
                           :authority_level collection_authority_level
                           :type            collection_type}
          :scores          all-scores)
-        (update :dataset_query #(some-> % json/parse-string mbql.normalize/normalize))
+        (update :dataset_query (fn [dataset-query]
+                                 (when-let [query (some-> dataset-query json/parse-string)]
+                                   (if (get query "type")
+                                      (mbql.normalize/normalize query)
+                                      (not-empty (lib/normalize query))))))
```
Comment (metamben) [inline src/metabase/search/scoring.clj:329]: I expect this logic will be needed in a bunch of places.

**Review by metamben — APPROVED** [review]
Nice!

**Review by kamilmielnik — COMMENTED** [review]
FE assumes that `database_id` & `table_id` attributes will be present in each metric search result.
Please test clicking on a metric in search results before merging this PR (currently it does not work because the aforementioned attributes are coming as `null`).

**Pull request merged**